js/bare_board.js

```diff
@@ -1,61 +1,46 @@
 import _ from 'lodash';
 import d3 from 'd3';
-import stringify from 'json-stringify-pretty-compact';
 
-import {tileSize, repositionSpeed, DEV_MODE, animationStepDuration, animationStepDurationMin, animationStepDurationMax, margin, stockColumns} from './config';
-import * as config from './config';
+import {tileSize, DEV_MODE, animationStepDuration, margin, stockColumns} from './config';
 import {CanvasParticleAnimation} from './particle/canvas_particle_animation';
 import * as simulation from './simulation';
-import {Stock} from './stock';
 import * as tile from './tile';
-import {Level} from './level';
 import {WinningStatus} from './winning_status';
 import {bindDrag} from './drag_and_drop';
 import {Logger} from './logger';
 import {SoundService} from './sound_service';
 
-function tileSimpler(name, i, j) {
-  const tileClass = tile[name];
-  return new tile.Tile(tileClass, 0, false, i, j);
-}
-
-export class Board {
-  constructor(level, svg, helper, titleManager, levels, progressPearls, storage) {
-    this.levels = levels;
-    this.levelsLookup = _.indexBy(levels, (levelRecipe) => `${levelRecipe.group} ${levelRecipe.name}`);
+export class BareBoard {
+  constructor(svg, callbacks = {}) {
     this.svg = svg;
     this.tileMatrix = [];
-    this.helper = helper;
-    this.titleManager = titleManager;
-    this.progressPearls = progressPearls;
-    this.storage = storage;
     this.animationStepDuration = animationStepDuration;
-    this.stock = new Stock(svg, this);
+
+    // NOTE maybe some event listener instead?
+    this.callbacks = {};
+    this.callbacks.experimentDisturbed = callbacks.experimentDisturbed || (() => null);
+    this.callbacks.tileRotated = callbacks.tileRotated || (() => null);
+    this.callbacks.tileMouseover = callbacks.tileMouseover || (() => null);
+    this.callbacks.animationStart = callbacks.animationStart || (() => null);
+    this.callbacks.animationEnd = callbacks.animationEnd || (() => null);
+
     this.logger = new Logger();
     this.logger.logAction('initialLevel');
-    // Load the level
-    this.loadLevel(level);
+
+    // this field is modified by ParticleAnimation
+    this.animationExists = false;
```

Comment:
Nie za bardzo rozumiem na czym zyskujemy dorabiając to pole. W dodatku nazwa jest nieco myląca, bo po wciśnięciu play a potem pause, `this.particleAnimation` jest instancją `ParticleAnimation`, natomiast `this.animationExists === false`.

Chodzi mi o to, że nie widzę jakiejś istotnej zmiany w efektywnym sposobie działania (nie widzę, żeby to był fix czegokolwiek), a wręcz powiększamy utrzymywany stan.